### PR TITLE
Checkstyle config and task in ./fluxc/

### PR DESCRIPTION
**Pull request by maxme**
To test, run:

```shell
./gradlew checkstyle
```

Notices all the errors and cry.

Note that you can install the checkstyle-idea plugin in AS and import the checkstyle.xml config from `config/checkstyle.xml`

### CODE REVIEW

**Comment by aforcier** [comment]
Looks good! Tried this out with the plugin too, works nicely.

What about enforcing:
> self hosted vs dot org and wpcom vs dotcom and probably use checkstyle to enforce them

(from https://github.com/wordpress-mobile/WordPress-FluxC-Android/issues/14#issuecomment-196312541)

**Comment by maxme** [comment]
> What about enforcing:

Yes, I'll try to add a regexp rule for that.

**Comment by maxme** [comment]
> Yes, I'll try to add a regexp rule for that.

Done in  	5a3add7

File: config/checkstyle.xml [hunk 1]
```diff
@@ -51,6 +51,22 @@
       <property name="severity" value="error"/>
     </module>
 
+    <!-- Specific to FluxC / WordPress -->
+    <module name="RegexpSingleline">
+      <property name="format" value="dotcom"/>
+      <property name="ignoreCase" value="true"/>
+      <property name="message" value="Kim is not happy, you should not use _dotcom_, use _wpcom_ instead"/>
```
Comment (aforcier) [inline config/checkstyle.xml]: 😆

**Comment by aforcier** [comment]
:shipit:

**Pull request merged**